### ISSUE TITLE
Error in seq.default

### ISSUE THREAD

**Issue by jing-wan** [dphansti/mango#19]
Hi, 
I meet some errors when I used mango: 
`[1] "aligning reads"
Error in seq.default(1, length(lines), 4) : wrong sign in 'by' argument
Calls: alignBowtie -> findScore -> paste -> seq -> seq.default
Execution halted
`
But I don't what's the reason of it. 
Here are my command:
`Rscript mango.R --fastq1 ~SRR1296617_1.fastq --fastq2 ~SRR1296617_2.fastq --prefix SRR1296617 --argsfile argsfile.txt --outdir ~/mango/SRR1296617`

my argsfile.txt is  :
`bowtieref = ~/data/ref_hg19_UCSC/hg19_bowtie2

bedtoolsgenome = ~/data/hg19.chrom.sizes`

I don't know whether is the error of my argsfile.txt . What should be the correct bowtieref and bedtoolsgenome?

Thank you!

**Comment by dphansti** [comment]
Currently Mango only supports bowtie (not bowtie2).  So make sure bowtie is in you PATH (or specify the path to it in your argsfile).  And set bowtieref to a bowtie1 reference.  See if that fixes the problem.

**Comment by jing-wan** [comment]
@dphansti 
I changed to bowtie1 just now, but it's the same error:
`Rscript mango.R --fastq1 ~SRR1296617_1.fastq --fastq2 ~SRR1296617_2.fastq --prefix SRR1296617 --argsfile argsfile.txt --outdir ~/mango/SRR1296617`
`[1] "aligning reads"
Error in seq.default(1, length(lines), 4) : wrong sign in 'by' argument
Calls: alignBowtie -> findScore -> paste -> seq -> seq.default
Execution halted
`
And my argsfile.txt is:
```
bowtieref = ~/data/hg19_UCSC_bowtie/hg19_bowtie
bedtoolsgenome = ~/data/hg19.chrom.sizes

```

`~/data/hg19_UCSC_bowtie/` is my bowtie index path, and `hg19_bowtie` is the Prefix of bowtie index. 
And` ~/data/hg19.chrom.sizes` is a file include the size of genome chromesomes.

The bowtieref should be a PATH or PATH+Prefix  or a fasta file?

**Comment by IoannisVardaxis** [comment]
Hi,
I have exactly the same problem. I also use Bowtie. Have you found any solution?
I am using Mango on three different datasets, two of them run perfectly, however I get the above error on the other one. 

I noticed however that for that dataset if I set keepempty=TRUE, then it will run the bowtie step but it crashes on another step again. 
The thing is that since it runs the bowtie step with keepempty=TRUE, it means that the linker sequences are actually not found on any of the alignments? And the thing is that I dont really understand if this is a problem of Mango or if the dataset is problematic. The linker sequences should be the default ones from Mango because the data is supposed to be generated using MmeI. 

So the error I get after setting keepempty=TRUE is the following:

[1] "building bedpe"
Error: read names of PET ends do not match[1] "removing duplicate PETs"
Error: Column index must be at most 0 if positive, not 5
Execution halted

The dataset can be found here:  https://www.encodeproject.org/experiments/ENCSR000BZZ/
Three replicates can be found for the dataset, I have tried all of them and all crash.


Thank you

**Comment by IoannisVardaxis** [comment]
Hey again,

I have a little update. I tried to run Mango step by step manually for the dataset that doesnt work. 
I found out that the error is actually not in bowtie. At stage 1 of Mango when the following command runs : 

print ("finding linkers")

  parsingresults = parseFastq( fastq1=fastq1,
              fastq2=fastq2,
              basename = basename,
              minlength = minlength,
              maxlength = maxlength,
              keepempty = keepempty,
              linker1=linker1,
              linker2=linker2,
              numberlinkers=numberlinkers)

then the outputs from this which are saved in the directory have zero bytes. That is they are empty.
The hash states it too:

<hash> containing 4 key-value pair(s).
  ambigious PETs : 26170257
  chimeric PETs : 0
  same PETs : 0
  total PETs : 26170257

So because of that, bowtie cannot align anything I assume. 

Furthermore, I went into MangoC.cpp script and run the parseFastq function from there. 
It turns out that both r1linker and r2linker variables take always the value 0. And given that keepempty==FALSE, the PETs are ambigious. So the problem is actually that the input linker sequences are not found in the data, right? However they should be there since the data is from Mmel, strange.

**Comment by jing-wan** [comment]
Yes, I encountered this problem because my data don't have linker. 
So I skipped the stage 1 and changed the fastq filename to .same.fastq as my input.

**Comment by IoannisVardaxis** [comment]
Hey,

Thanks for the update. I will try what you say and see what happens. It seems my data does not have linker either. However I have the raw unprocessed data and it should have linkers. I guess the data is poorly documented. 

However if I do what you say, i will use all the PETs in the interaction analysis, and no classification on ambigious/chimeric/same will take place. If the data (mine and yours too) contain all of those, then your interactions might not be so trustworthy. Unless you know that your data contains only the correct type of PETs. Mine does not give any such information, they dont say that linkers are removed to begin with...
